kornia/augmentation/_2d/intensity/gaussian_illumination.py

```diff
@@ -0,0 +1,156 @@
+                )
+        else:
+            raise ValueError("gain must be a tuple or a float")
+        KORNIA_CHECK(
+            all(0 <= el <= 1 for el in gain),
+            "gain values must be between 0 and 1. Recommended values less than 0.2.",
+        )
+
+        if isinstance(center, (tuple, float)):
+            if isinstance(center, float):
+                center = (center, center)
+            elif len(center) == 1:
+                center = (center[0], center[0])
+            elif len(center) > 2 or len(center) <= 0:
+                raise ValueError(
+                    "The length of center must be greater than 0 \
+                        and less than or equal to 2, and it should be a tuple or a float."
+                )
+        else:
+            raise ValueError("center must be a tuple or a float")
+        KORNIA_CHECK(
+            all(0 <= el <= 1 for el in center),
+            "center of gaussian value must be between 0 and 1.",
+        )
+
+        if isinstance(sigma, (tuple, float)):
+            if isinstance(sigma, float):
+                sigma = (sigma, sigma)
+            elif len(sigma) == 1:
+                sigma = (sigma[0], sigma[0])
+            elif len(sigma) > 2 or len(sigma) <= 0:
+                raise ValueError(
+                    "The length of sigma must be greater than 0 \
+                        and less than or equal to 2, and it should be a tuple or a float."
+                )
+        else:
+            raise ValueError("sigma must be a tuple or a float")
+        KORNIA_CHECK(
+            all(0 <= el <= 1 for el in sigma),
+            "sigma of gaussian value must be between 0 and 1.",
+        )
+
+        if isinstance(sign, (tuple, float)):
+            if isinstance(sign, float):
+                sign = (sign, sign)
+            elif len(sign) == 1:
+                sign = (sign[0], sign[0])
+            elif len(sign) > 2 or len(sign) <= 0:
+                raise ValueError(
+                    "The length of sign must be greater than 0 \
+                        and less than or equal to 2, and it should be a tuple or a float."
+                )
+        else:
+            raise ValueError("sign must be a tuple or a float")
+        KORNIA_CHECK(
+            all(-1 <= el <= 1 for el in sign),
+            "sign of gaussian value must be between -1 and 1.",
+        )
+
+        # Generator of random parameters and masks.
+        self._param_generator = GaussianIlluminationGenerator(gain, center, sigma, sign)
+
+    def apply_transform(
+        self,
+        input: Tensor,
+        params: Dict[str, Tensor],
+        flags: Dict[str, Any],
+        transform: Optional[Tensor] = None,
+    ) -> Tensor:
+        r"""Apply random gaussian gradient illumination to the input image."""
+        KORNIA_CHECK(len(input.shape) in (3, 4), "Wrong input dimension.")
+        if len(input.shape) == 3:
+            input = input[None, :, :, :]
```

Comment:
it's automatically handled by the base API, so you don't need to check or try to handle the input shape

```suggestion
```